### PR TITLE
Replace deprecated nrfx API

### PR DESCRIPTION
**Pull request by awojasinski-nordicsemi**
With the release of nrfx2.9 some API become deprecated
(see https://github.com/NordicSemiconductor/nrfx/blob/v2.9.0/CHANGELOG.md).
This PR replaces deprecated API with suggested in CHANGELOG

### CODE REVIEW

**Comment by awojasinski-nordicsemi** [comment]
I'm unsure if anything should be added to the changelog as the introduced changes are not affecting end-users and are internal changes.

**Comment by joerchan** [comment]
To do this properly then sdk-zephyr and sdk-trusted-firmware-m should be updated in addition. Are you gonna do that as well?

File: subsys/mpsl/cx/generic_3pin/mpsl_cx_generic_3pin.c [hunk 1]
```diff
@@ -285,17 +285,17 @@ static int mpsl_cx_init(const struct device *dev)
...
 	uint8_t pri_dir_pin = NRF_DT_GPIOS_TO_PSEL(CX_NODE, pri_dir_gpios);
 
-	soc_secure_gpio_pin_mcu_select(pri_dir_pin, NRF_GPIO_PIN_MCUSEL_NETWORK);
+	soc_secure_gpio_pin_mcu_select(pri_dir_pin, NRF_GPIO_PIN_MCU_NETWORK);
 #endif
 #if DT_NODE_HAS_PROP(CX_NODE, grant_gpios)
 	uint8_t grant_pin = NRF_DT_GPIOS_TO_PSEL(CX_NODE, grant_gpios);
 
-	soc_secure_gpio_pin_mcu_select(grant_pin, NRF_GPIO_PIN_MCUSEL_NETWORK);
+	soc_secure_gpio_pin_mcu_select(grant_pin, NRF_GPIO_PIN_MCU_NETWORK);
```
Comment (joerchan) [inline subsys/mpsl/cx/generic_3pin/mpsl_cx_generic_3pin.c:298]: `soc_secure_gpio_pin_mcu_select` in sdk-zephyr should be updated as well.

```
#if defined(GPIO_PIN_CNF_MCUSEL_Msk)
static inline void soc_secure_gpio_pin_mcu_select(uint32_t pin_number,
						  nrf_gpio_pin_mcusel_t mcu)
{
	nrf_gpio_pin_mcu_select(pin_number, mcu);
}
#endif /* defined(GPIO_PIN_CNF_MCUSEL_Msk) */
```

`#if defined(GPIO_PIN_CNF_MCUSEL_Msk)` and `nrf_gpio_pin_mcusel_t`

**Comment by awojasinski-nordicsemi** [comment]
@joerchan here is PR to zephyr: https://github.com/zephyrproject-rtos/zephyr/pull/49581

**Comment by joerchan** [comment]
Great, will you take it into this PR with a nrf fromtree patch once it is merged upstream then?

**Comment by awojasinski-nordicsemi** [comment]
> Great, will you take it into this PR with a nrf fromtree patch once it is merged upstream then?

PR fromlist : https://github.com/nrfconnect/sdk-zephyr/pull/896

**Review by SebastianBoe — APPROVED** [review]

**Comment by SebastianBoe** [comment]
LGTM. But needs rebase and green CI before the manifest child can be merged.

**Review by joerchan — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]
lgtm

**Comment by carlescufi** [comment]
@awojasinski-nordicsemi please rebase

**Comment by awojasinski-nordicsemi** [comment]
@carlescufi done

File: subsys/mpsl/cx/nrf700x/mpsl_cx_nrf700x.c [hunk 2]
```diff
@@ -285,7 +285,7 @@ static int mpsl_cx_init(const struct device *dev)
 #if DT_NODE_HAS_PROP(CX_NODE, req_gpios)
 	uint8_t req_pin = NRF_DT_GPIOS_TO_PSEL(CX_NODE, req_gpios);
 
-	soc_secure_gpio_pin_mcu_select(req_pin, NRF_GPIO_PIN_MCUSEL_NETWORK);
+	soc_secure_gpio_pin_mcu_select(req_pin, NRF_GPIO_PIN_MCU_NETWORK);
```
Comment (jciupis) [inline subsys/mpsl/cx/nrf700x/mpsl_cx_nrf700x.c:288]: Shouldn't it be `NRF_GPIO_PIN_SEL_NETWORK`? https://github.com/NordicSemiconductor/nrfx/blob/v2.9.0/hal/nrf_gpio.h#L169

Applies to other occurrences here and in `subsys/mpsl/cx/thread/mpsl_cx_thread.c`.

**Review by jciupis — CHANGES_REQUESTED** [review]

File: subsys/mpsl/cx/nrf700x/mpsl_cx_nrf700x.c (hunk 2 above)
Comment (awojasinski-nordicsemi) [reply subsys/mpsl/cx/nrf700x/mpsl_cx_nrf700x.c:288]: nope, as you can see in the description

> @note This enumerator is deprecated. Use @ref nrf_gpio_pin_sel_t instead

Comment (jciupis) [reply subsys/mpsl/cx/nrf700x/mpsl_cx_nrf700x.c:288]: Exactly. [The source code for `nrf_gpio_pin_sel_t` type](https://github.com/NordicSemiconductor/nrfx/blob/v2.9.0/hal/nrf_gpio.h#L152-L159) :
```c
/** @brief Enumerator used for selecting the MCU/Subsystem to control the specified pin. */
typedef enum
{
    NRF_GPIO_PIN_SEL_APP        = GPIO_PIN_CNF_MCUSEL_AppMCU,     ///< Pin controlled by Application MCU.
    NRF_GPIO_PIN_SEL_NETWORK    = GPIO_PIN_CNF_MCUSEL_NetworkMCU, ///< Pin controlled by Network MCU.
    NRF_GPIO_PIN_SEL_PERIPHERAL = GPIO_PIN_CNF_MCUSEL_Peripheral, ///< Pin controlled by dedicated peripheral.
    NRF_GPIO_PIN_SEL_TND        = GPIO_PIN_CNF_MCUSEL_TND,        ///< Pin controlled by Trace and Debug Subsystem.
} nrf_gpio_pin_sel_t;
```
That's why I suggested to use `NRF_GPIO_PIN_SEL_NETWORK` in this line.

Comment (awojasinski-nordicsemi) [reply subsys/mpsl/cx/nrf700x/mpsl_cx_nrf700x.c:288]: Ahh right, I quickly look at this failure and miss that is still deprecated enum used. Thanks for pointing out ;)

**Review by jciupis — APPROVED** [review]

**Comment by awojasinski-nordicsemi** [comment]
@carlescufi could you please merge that?

**Pull request merged**